ludwig/datasets/mnist/__init__.py

```diff
@@ -47,6 +47,18 @@ class Mnist(CSVLoadMixin, GZipDownloadMixin, BaseDataset):
     processed_dataset_path: str
 
     def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION):
+        try:
```

Comment:
Curious about importing `torchvision` during initialization of this class, instead of importing it directly with the other imports and import guarding it?

Reply:
the point is that if you import it directly you won0t get the message with what to do to fix the issue if the image dependencies are not installed, you would just get an import error

Reply:
This is true, but calling `sys.exit` is not very good for the user, as it could, for example, crash their IPython kernel. Better to raise a new `ImportError` with the more descriptive error message.

Reply:
That's a good point, although the same pattern is used in other parts of the code where we do this safeguarded import, so maybe we can have another PR that replaces the system exit with import errors all around.